### DISCUSSION TITLE
Changing numeric text field turns it into a string

### DISCUSSION THREAD

**Discussion by fdinhSG** [Kitware/trame#280] [Q&A] [answered]
Take the following script:

```python
from trame.app import get_server
from trame.ui.vuetify import VAppLayout
from trame.widgets import vuetify

server = get_server()
state, ctrl = server.state, server.controller

with VAppLayout(server) as layout:
    with vuetify.VContainer():
        with vuetify.VRow():
            with vuetify.VCol():
                vuetify.VTextField(label='number1', v_model=('number1', 1), type='number')
            with vuetify.VCol():
                vuetify.VTextField(label='number2', v_model=('number2', 1), type='number')
            with vuetify.VCol():
                vuetify.VBtn("sum", click=lambda: print(f'The sum is {state.number1 + state.number2}'))

server.start()
```

I have two numeric field and when I press the 'sum' button it prints the sum between the two.

However, as soon as I change one of the two fields and I try to sum them, the following error message appears:

    ERROR:root:TypeError("unsupported operand type(s) for +: 'int' and 'str'")

Meaning that the number has turned into a string.

Is this an intended effect? Is it possible to make it so that when the field changes, it remains a number?

Thanks!

**Comment by jourdain** [comment]
This is an expected behavior from such UI. To force them to be number, you can use [`v-model.number="..."`](https://v2.vuejs.org/v2/guide/forms.html#number). But with trame, we did not exposed it by default so you will have to do something like that

```python
vuetify.VTextField(
   label='number1', 
   v_model_number=('number1', 1), 
   type='number',
   __properties=[
      ("v_model_number", "v-model.number"),
   ],
)
```

**Reply by fdinhSG** [reply]
Thank you! Unfortunately it seems like it's not working...

I think I could make it work with something like

```python
@state.change('number1')
async def change_detected(number1, **kwargs):
    state.number1 = int(number1)
```

Far from elegant, but it works.

**Comment by jourdain** [comment, marked as answer]
The following is actually working for me

```python
from trame.app import get_server
from trame.ui.vuetify import VAppLayout
from trame.widgets import vuetify

server = get_server()

@server.state.change("value")
def update_number(value, **kwargs):
    print(f"{type(value)}({value})")

with VAppLayout(server):
    vuetify.VTextField(v_model_number=('value', 1), type='number')

server.start()
```
